framework/sandstone_tests.h

```diff
@@ -11,20 +11,85 @@
 
 #ifdef __cplusplus
 #include <span>
+#include <cstring>
+#include <iterator>
+#include <map>
```

Comment:
Nitpick: alphabetical order.

`<string.h>` is already included and you're not using `std::` with any of the functions below, so the `<cstring>` is superfluous.